WooCommerce/Classes/POS/Presentation/Card Present Payments/PointOfSaleCardPresentPaymentEventPresentationStyle.swift

```diff
@@ -0,0 +1,114 @@
+
+        case .connectingFailed(let error, let retrySearch, let endSearch):
+            return .alert(.connectingFailed(
+                viewModel: PointOfSaleCardPresentPaymentConnectingFailedAlertViewModel(
+                    error: error,
+                    retryButtonAction: retrySearch,
+                    cancelButtonAction: endSearch)))
+
+        case .connectingFailedNonRetryable(let error, let endSearch):
+            return .alert(.connectingFailedNonRetryable(
+                viewModel: PointOfSaleCardPresentPaymentConnectingFailedNonRetryableAlertViewModel(
+                    error: error,
+                    cancelAction: endSearch)))
+
+        case .connectingFailedUpdatePostalCode(let retrySearch, let endSearch):
+            return .alert(.connectingFailedUpdatePostalCode(
+                viewModel: PointOfSaleCardPresentPaymentConnectingFailedUpdatePostalCodeAlertViewModel(
+                    retryButtonAction: retrySearch,
+                    cancelButtonAction: endSearch)))
+
+        case .connectingFailedChargeReader(let retrySearch, let endSearch):
+            return .alert(.connectingFailedChargeReader(
+                viewModel: PointOfSaleCardPresentPaymentConnectingFailedChargeReaderAlertViewModel(
+                    retryButtonAction: retrySearch,
+                    cancelButtonAction: endSearch)))
+
+        case .connectingFailedUpdateAddress(let wcSettingsAdminURL, let retrySearch, let endSearch):
+            return .alert(.connectingFailedUpdateAddress(
+                viewModel: PointOfSaleCardPresentPaymentConnectingFailedUpdateAddressAlertViewModel(
+                    settingsAdminUrl: wcSettingsAdminURL,
+                    retrySearchAction: retrySearch,
+                    cancelSearchAction: endSearch)))
+
+        case .foundReader(let name, let connect, let continueSearch, let endSearch):
+            return .alert(.foundReader(
+                viewModel: PointOfSaleCardPresentPaymentFoundReaderAlertViewModel(
+                    readerName: name,
+                    connectAction: connect,
+                    continueSearchAction: continueSearch,
+                    endSearchAction: endSearch)))
+
+        case .updateProgress(let requiredUpdate, let progress, let cancelUpdate):
+            return .alert(.updatingReader(viewModel: CardPresentPaymentUpdatingReaderAlertViewModel()))
+
+        case .updateFailed(let tryAgain, let cancelUpdate):
+            return .alert(.updateFailed(viewModel: CardPresentPaymentReaderUpdateFailedAlertViewModel()))
+
+        case .updateFailedNonRetryable(let cancelUpdate):
+            return .alert(.updateFailed(viewModel: CardPresentPaymentReaderUpdateFailedAlertViewModel()))
+
+        case .updateFailedLowBattery(let batteryLevel, let cancelUpdate):
+            return .alert(.updateFailed(viewModel: CardPresentPaymentReaderUpdateFailedAlertViewModel()))
```

Comment:
non-blocking: might we want to mention the low battery info in the alert? Feel free to ignore this if you already have this as a future task.

Reply:
Yep, I'll tackle all the update details in a later task, they seemed lower priority than getting the (normal) connection flow working and this PR is big enough!